### ISSUE TITLE
Task `initializeSonatypeStagingRepository` fails when using `gradle-setup` action

### ISSUE THREAD

**Issue by nsenave** [gradle-nexus/publish-plugin#347]
Hi, thanks for your plugin which helps a lot.

I use the plugin in a GitHub workflow which can be summarized like this:

```yaml
    steps:
      - uses: actions/checkout@v4
      - uses: actions/setup-java@v4
        # with: env variables and secrets 
      - uses: gradle/gradle-build-action@v3
      - run: ./gradlew clean publishToSonatype closeAndReleaseSonatypeStagingRepository
        # with: env variables and secrets
```

Works fine. 

I've recently replaced `gradle/gradle-build-action@v3` (which is deprecated) with `gradle/actions/setup-gradle@v3`.

Then, I have this error: 

```
Run ./gradlew clean publishToSonatype closeAndReleaseSonatypeStagingRepository
Starting a Gradle Daemon (subsequent builds will be faster)
> Task :model:clean UP-TO-DATE

> Task :initializeSonatypeStagingRepository FAILED
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':initializeSonatypeStagingRepository'.
2 actionable tasks: 1 executed, 1 up-to-date
> Failed to load staging profiles, server at https://oss.sonatype.org/service/local/ responded with status code 403, body: <html>
    <head>
      <title>403 - Forbidden</title>
      <meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
  
      <link rel="icon" type="image/png" href="https://oss.sonatype.org/favicon.png">
      <!--[if IE]>
      <link rel="SHORTCUT ICON" href="https://oss.sonatype.org/favicon.ico"/>
      <![endif]-->
  
      <link rel="stylesheet" href="https://oss.sonatype.org/static/css/Sonatype-content.css?2.15.1-02" type="text/css" media="screen" title="no title" charset="utf-8">
    </head>
    <body>
      <h1>403 - Forbidden</h1>
      <p>Forbidden</p>
    </body>
  </html>
```

Gradle wrapper version used: 8.6

Actions `gradle-build-action` and `actions/setup-gradle` [are supposed to be functionally identical](https://github.com/gradle/actions/blob/main/docs/deprecation-upgrade-guide.md), so it might be an issue on their side in the end.

Yet, I thought you might have an idea / ways to reproduce the bug and diagnose why the error happens.

**Comment by joffrey-bion** [comment]
I got the same error, but I think that has to do with Sonatype changing its authentication system. Did you check that you can still login on https://oss.sonatype.org/?

**Comment by szpak** [comment]
I would also recommend generating the new token. Some other people were reporting the similar problem on the Sonatype mailing list recently.

**Comment by nsenave** [comment]
Thanks for the quick answers. I should have precised that I managed to publish a few minutes after receving the 403 error by using back the "gradle-build-action".

A sonatype login could look coherent indeed, yet it seems that it doesn't match in my case:

- Jul 12, at morning: the user token was changed
- Jul 12, afternoon: a publication succeeded (using `gradle-build-action@v3` in the workflow)
- Jul 19: publication fails (using `actions/setup-gradle@v3`)
- Jul 19, few minutes after, using `gradle-build-action@v3` back: publication succeeded

Seems like something weird happens with caching, but I'm having a hard time understanding what that could be: maybe definitely an issue for the [gradle setup](https://github.com/gradle/actions/tree/main/setup-gradle) action.

**Comment by joffrey-bion** [comment]
I'm getting the same 401s now. Any news on this?

**Comment by szpak** [comment]
@joffrey-bion Is this still happening after 12-24 hours?

**Comment by joffrey-bion** [comment]
Yes, the first occurrence was about 4 days ago, and I'm still facing the issue in multiple repositories. For example:

https://github.com/joffrey-bion/socketio-kotlin/actions/workflows/release.yml

**Comment by szpak** [comment]
Have you re-generated your Nexus token (and updated in in GH)?

**Comment by joffrey-bion** [comment]
Yes, sorry not to have mentioned this more clearly. I generated a new token and updated my repo secrets with it.

**Comment by joffrey-bion** [comment]
It looks like for some repos the failure correlates with the upgrade to Gradle 8.10, but that might be a red herring, because at least one of my repos could be published with Gradle 8.10 and setup-gradle v4.

**Comment by joffrey-bion** [comment]
Ok sorry for the noise about this, the 401s were caused because of unfortunate naming of secrets, and I had swapped the username and password. All good now.

**Comment by szpak** [comment]
@joffrey-bion Thanks for the update!

@nsenave Have you manager to solve the problem with your repo?

**Comment by nsenave** [comment]
Situation didn't change for me since my [last comment](https://github.com/gradle-nexus/publish-plugin/issues/347#issuecomment-2242540215), I'll give it another try and keep you informed.

**Comment by Ahmedhossamdev** [comment]
Same issue happened with me; I'm sure that all my secrets are correct!

https://github.com/OneBusAway/kotlin-sdk/actions/runs/11241615588/job/31254075229

Do you guys generate tokens from here?

https://central.sonatype.com/account

**Comment by joffrey-bion** [comment]
> Do you guys generate tokens from here?

No that's for the new Publisher Api. I think you probably still need to generate your token for OSSRH in the Nexus:
https://oss.sonatype.org/#welcome

You can log in in the top right, then click on your logged in username in the same place, and open your Profile.
Then, change the drop down (just below the tab headers) from `Summary` to `User Token`.
From there, you can regenerate a new token for the "old way" of publishing.

**Comment by nsenave** [comment]
Hi, sorry it's been a while. I've recently uploaded a new artifact of my project using the plugin. 

Worked well in combination with the latest version of the `actions/setup-gradle` (`v6`).

Note: my project was configured to deploy on the legacy sonatype server `https://oss.sonatype.org/service/local/`, I had to replace it with the new url `https://ossrh-staging-api.central.sonatype.com/service/local/`

Sonatype docs: https://central.sonatype.org/publish/publish-portal-ossrh-staging-api/#configuring-your-plugin

TL;DR: to me the issue is solved! ✅

**Comment by szpak** [comment]
Thanks @nsenave for sharing your story and what helped you deal with the problem!
